File: src/main/java/com/example/travelator/_11methodtoproperty/v10/kotlin/CampSites.java
package com.example.travelator._11methodtoproperty.v10.kotlin;

import java.util.Set;
import java.util.stream.Collectors;

public class CampSites {
	
	public static Set<CampSite> sitesInRegion(
		Set<CampSite> sites,
		String countryISO,
		String region
	) {
		return sites.stream()
				.filter(campSite ->
					campSite.getCountryCode().equals(countryISO)
					&& campSite.getRegion().equalsIgnoreCase(region)
				)
				.collect(Collectors.toUnmodifiableSet());
	}
}

File: src/main/java/com/example/travelator/_11methodtoproperty/v10/kotlin/CampSites.kt
package com.example.travelator._11methodtoproperty.v10.kotlin

import java.util.stream.Collectors

object CampSites {
    fun sitesInRegion(
        sites: Set<CampSite>,
        countryISO: String,
        region: String?
    ): Set<CampSite> {
        return sites.stream()
            .filter { campSite: CampSite ->
                campSite.countryCode == countryISO
                        && campSite.region.equals(region, ignoreCase = true)
            }
            .collect(Collectors.toUnmodifiableSet())
    }
}